### PR TITLE
TargetExtension Unittests

### PR DESCRIPTION
**Pull request by PravinPK**

### CODE REVIEW

File: code/android-target-library/src/test/java/com/adobe/marketing/mobile/target/TargetExtensionTests.java [hunk 1]
```diff
@@ -74,5 +222,2147 @@ public void test_getVersion() {
...
+    // readyForEvent tests
+    //**********************************************************************************************
+
+    @Test
+    public void test_readyForEvent() {
+        // setup
+        setConfigurationSharedState(MOCKED_CLIENT_CODE, 5, "optedin", false, "targetserver");
+
+        // test
+        extension.readyForEvent(noEventDataEvent());
```
Comment (rymorale) [inline code/android-target-library/src/test/java/com/adobe/marketing/mobile/target/TargetExtensionTests.java:259]: as part of verification we could assertTrue here

File: code/android-target-library/src/test/java/com/adobe/marketing/mobile/target/TargetExtensionTests.java [hunk 2]
```diff
@@ -74,5 +222,2147 @@ public void test_getVersion() {
...
+
+        // verify target state is updated with correct configuration
+        verify(targetState).updateConfigurationSharedState(notNull());
+    }
+
+    @Test
+    public void test_readyForEvent_whenConfigurationIsNotSet() {
+        // test
+        extension.readyForEvent(noEventDataEvent());
```
Comment (rymorale) [inline code/android-target-library/src/test/java/com/adobe/marketing/mobile/target/TargetExtensionTests.java:268]: we could assertFalse here as well

File: code/android-target-library/src/test/java/com/adobe/marketing/mobile/target/TargetExtensionTests.java [hunk 3]
```diff
@@ -74,5 +222,2147 @@ public void test_getVersion() {
...
+
+        // verify that five listeners are registers
```
Comment (rymorale) [inline code/android-target-library/src/test/java/com/adobe/marketing/mobile/target/TargetExtensionTests.java:270]: typo: registered

File: code/android-target-library/src/test/java/com/adobe/marketing/mobile/target/TargetExtensionTests.java [hunk 4]
```diff
@@ -74,5 +222,2147 @@ public void test_getVersion() {
...
+    //**********************************************************************************************
+    // onRegister Tests
+    //**********************************************************************************************
+
+    @Test
+    public void test_onRegister() {
+        // test
+        extension.onRegistered();
+
+        // verify that five listeners are registers
```
Comment (rymorale) [inline code/android-target-library/src/test/java/com/adobe/marketing/mobile/target/TargetExtensionTests.java:245]: typo: registered

File: code/android-target-library/src/test/java/com/adobe/marketing/mobile/target/TargetExtensionTests.java [hunk 5]
```diff
@@ -74,5 +222,2147 @@ public void test_getVersion() {
...
+        // set privacy status optedOut
+        final HashMap<String, Object> emptySharedState = new HashMap();
+        when(targetState.getMobilePrivacyStatus()).thenReturn(MobilePrivacyStatus.OPT_OUT);
+        when(targetState.generateSharedState()).thenReturn(emptySharedState);
+
+        // test
+        final Event event = setThirdPartyIdEvent(MOCK_THIRD_PARTY_ID);
+        extension.handleTargetRequestIdentityEvent(event);
+
+        // validate that the shared state is updated with empty map and the thrid party Id is not set
```
Comment (rymorale) [inline code/android-target-library/src/test/java/com/adobe/marketing/mobile/target/TargetExtensionTests.java:695]: typo: third

**Review by rymorale — DISMISSED** [review]
looks good to me, just a couple of typos to fix

**Comment by PravinPK** [comment]
Thanks for the review Ryan!
 I will make the necessary changes and merge this PR

**Review by rymorale — APPROVED** [review]

**Pull request merged**